com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -8,101 +8,362 @@
 namespace Unity.Netcode
 {
     /// <summary>
-    /// Main class for managing network scenes
+    /// Used for local notifications of various scene events.
+    /// The <see cref="NetworkSceneManager.OnSceneEvent"/> of delegate type <see cref="NetworkSceneManager.SceneEventDelegate"/> uses this class to provide
+    /// scene event status/state.
     /// </summary>
-    public class NetworkSceneManager
+    public class SceneEvent
     {
         /// <summary>
-        /// Delegate for when the scene has been switched
+        /// If applicable, this will be set to the <see cref="UnityEngine.AsyncOperation"/> returned by <see cref="SceneManager"/>
+        /// load scene and unload scene asynchronous methods.
         /// </summary>
-        public delegate void SceneSwitchedDelegate();
+        public AsyncOperation AsyncOperation;
 
         /// <summary>
-        /// Delegate for when a scene switch has been initiated
+        /// Will always be set to the current scene event type (<see cref="SceneEventData.SceneEventTypes"/>) this scene event notification pertains to
         /// </summary>
-        public delegate void SceneSwitchStartedDelegate(AsyncOperation operation);
+        public SceneEventData.SceneEventTypes SceneEventType;
 
         /// <summary>
-        /// Delegate for when a client has reported to the server that it has completed scene transition
-        /// <see cref='OnNotifyServerClientLoadedScene'/>
+        /// If applicable, this reflects the type of scene loading or unloading that is occurring.
+        /// Unlike <see cref="SceneManager"/>, scene unload events will have the original <see cref="LoadSceneMode"/> applied when the scene was loaded.
         /// </summary>
-        public delegate void NotifyServerClientLoadedSceneDelegate(SceneSwitchProgress progress, ulong clientId);
+        public LoadSceneMode LoadSceneMode;
```

Comment:
Nit, and I know it was like this, but if we really are talking just 'additive' vs. 'single' (and it seems to me those will be the only choices), this might be easier to reason about if it was a boolean "isAdditive"

Reply:
This is part of the SceneEvent that users receive if they have subscribed to the OnSceneEvent.  If we changed this to a bool then we would force users to have to make that adjustment as well and it would deviate from the way Unity identifies this type of scene loading.  If a user calls NetworkSceneManager.LoadScene and passes in LoadSceneMode.Additive then they should expect the notification to return that same value type when they receive the notification the scene has started to load, the scene is loaded (locally), and all clients have loaded the scene.